### PR TITLE
disconnect miners when pool is not operational

### PR DESCRIPTION
**Pull request by xrdavies**
* auto switch pool if the current pool is down
* disconnect miners when set pool not operational

### CODE REVIEW

**Comment by jonano614** [comment]
I suggest to leave that feature for the next version.

**Comment by jonano614** [comment]
I changed base branch to develop. Resolve conflict please.

**Comment by xrdavies** [comment]
> I changed base branch to develop. Resolve conflict please.

Resolved.

File: client/pool.c [hunk 1]
```diff
@@ -495,6 +498,7 @@ void *pool_net_thread(void *arg)
 	socklen_t peeraddr_len = sizeof(peeraddr);
 	int rcvbufsize = 1024;
 
+begin:
```
Comment (jonano614) [inline client/pool.c]: Oh, please, don't use goto. It is bad code style.

File: client/init.c [hunk 2]
```diff
@@ -189,6 +189,8 @@ int xdag_init(int argc, char **argv, int isGui)
 		return -1;
 	}
 
+	g_xdag_auto_pick_pool = !mining_threads_count ? 1 : (!is_pool && pool_arg == NULL); /* auto pick pool */
```
Comment (jonano614) [inline client/init.c]: It seems g_xdag_auto_pick_pool  is always 1.
xdag  is started as a pool - mining_threads_count is 0 and g_xdag_auto_pick_pool  = 1
xdag is started as a wallet without mining - mining_threads_count is 0 and g_xdag_auto_pick_pool = 1
xdag is started as a wallet with mining - mining_threads_count is > 0 but (!is_pool && pool_arg == NULL)  is true
And why do you need (!is_pool && pool_arg == NULL)?

**Pull request closed without merging**